src/autopas/utils/logging/FLOPLogger.cpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file FLOPLogger.cpp
+ * @author F. Gratl & S. Newcome
+ * @date 17/04/2024
+ */
+
+#include "FLOPLogger.h"
+
+#include <string>
+
+#include "Logger.h"
+#include "autopas/utils/Timer.h"
+
+autopas::FLOPLogger::FLOPLogger(const std::string &outputSuffix) : _loggerName("FLOPLogger" + outputSuffix) {
+#ifdef AUTOPAS_LOG_FLOPS
+  _loggedEmptyFields = false;
+  const auto *fillerAfterSuffix = outputSuffix.empty() or outputSuffix.back() == '_' ? "" : "_";
+  const auto outputFileName("AutoPas_FLOPCount_" + outputSuffix + fillerAfterSuffix +
+                            autopas::utils::Timer::getDateStamp() + ".csv");
+  // Start of workaround: Because we want to use an asynchronous logger we can't quickly switch patterns for the header.
+  // Create and register a non-asychronous logger to write the header.
+  const auto headerLoggerName = _loggerName + "header";
+  auto headerLogger = spdlog::basic_logger_mt(headerLoggerName, outputFileName);
+  // set the pattern to the message only
+  headerLogger->set_pattern("%v");
+  // print csv header
+  headerLogger->info(
+      "Date,"
+      "Iteration,"
+      "Number FLOPs,"
+      "Hit Rate");
+  spdlog::drop(headerLoggerName);
+  // End of workaround
+
+  // create and register the actual logger
+  auto logger = spdlog::basic_logger_mt<spdlog::async_factory>(_loggerName, outputFileName);
+  // set pattern to provide date
+  logger->set_pattern("%Y-%m-%d %T,%v");
+#endif
+}
+
+autopas::FLOPLogger::~FLOPLogger() {
+#ifdef AUTOPAS_LOG_FLOPS
+  if (_loggedEmptyFields) {
+    AutoPasLog(INFO,
```

Comment:
1. I think this should be a warning. It's not spammy and if the user is annoyed by it they should disable the feature they did not implement because that is improper utilization.
2. Maybe also write this message the first time this happens otherwise a user will only read this at the end of their simulation.
3. I'm unsure if I would like this message only at the start, only the end, or both. Pro/Cons:
   - Start: Can be seen quickly after starting the simulation leading to better reaction times but easy to miss.
   - End: Harder to miss but very late.
   - Both: Best of both worlds but duplicated warning.

Reply:
You're a bit too premature with this comment haha - it wasn't ready for review and this doesn't really apply anymore.

FYI, this didn't actually work, I think, because the AutoPas logger gets deregistered before this object is destructed. So I opted with a much more heavy-handed approach of just logging "Not Implemented".

The Start doesn't work because we would need to force the user simulator to tell the logger which functors it will use (unless I am missing something). This could be problematic.

Option 2 could work, but might be problematic if this is in the depths of the log file and not that noticeable. As a `WARN` this might not be too bad.

I will post a comment summarising this further change.